File: 프로그래머스/unrated/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.cpp
#include <string>
#include <vector>
using namespace std;

struct coord {
    int x;
    int y;
};

coord dir[4] = {{-1, 0}, {0, -1}, {0, 1}, {1, 0}};

int solution(vector<vector<string>> board, int h, int w) {
    int answer = 0;
    for (int i = 0; i < 4; i++) {
        int z = h + dir[i].x, k = w + dir[i].y;
        if (z >= 0 && z < board[0].size() && k >= 0 && k < board.size())
            if (board[h][w] == board[z][k]) answer++;
    }
    return answer;
}

File: 프로그래머스/1/250125. ［PCCE 기출문제］ 9번 ／ 이웃한 칸/［PCCE 기출문제］ 9번 ／ 이웃한 칸.java
class Solution {
    
    static int[] dx = {0, 1, 0, -1};
    static int[] dy = {1, 0, -1, 0};
    
    public int solution(String[][] board, int h, int w) {
        int answer = 0;
        
        int n = board.length;
        int m = board[0].length;
        String color = board[h][w];
        for (int i = 0; i < 4; i++) {
            int nx = h + dx[i];
            int ny = w + dy[i];
            
            if (nx < 0 || ny < 0 || nx >= n || ny >= m) continue;
            if (board[nx][ny].equals(color)) answer++;
        }
        
        
        return answer;
    }
}